File: problems/0053/0053.cpp
#include <algorithm>
#include <vector>

class Solution {
public:
    int maxSubArray(std::vector <int>& nums) {
        int curr_max = nums[0];
        int global_max = nums[0];
        for (int i = 1; i < nums.size(); ++i) {
            curr_max = std::max(curr_max + nums[i], nums[i]);
            global_max = std::max(global_max, curr_max);
        }
        return global_max;
    }
};

File: problems/0001 - 0400/0053/0053.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr_max = global_max = nums[0]
        for i in range(1, len(nums)):
            curr_max = max(curr_max + nums[i], nums[i])
            global_max = max(global_max, curr_max)
        return global_max
